### PR TITLE
Address macOS issue where right-clicking a username in the main chat

### PR DESCRIPTION
**Pull request by ZeldaZach**
…(or game chat) areas would pop up a seemingly empty user profile. This is because the resize event is overridden and doesn't actually attempt to resize based on the size hint of the dialog. Now that we're explicit with the call, this resize should be forced and have comparable results to popping up user profile from the user list.

## Related Ticket(s)
- Fixes #IssueNumber

Before:
Screen Shot 2022-01-15 at 1 27 54 AM

After:
Screen Shot 2022-01-15 at 1 25 36 AM

### CODE REVIEW

File: cockatrice/src/userinfobox.cpp [hunk 1]
```diff
@@ -24,7 +24,7 @@ UserInfoBox::UserInfoBox(AbstractClient *_client, bool _editable, QWidget *paren
 {
     QFont nameFont = nameLabel.font();
     nameFont.setBold(true);
-    nameFont.setPointSize(nameFont.pointSize() * 1.5);
+    nameFont.setPointSize(std::ceil(nameFont.pointSize() * 1.5));
```
Comment (ebbit1q) [inline cockatrice/src/userinfobox.cpp:27]: qCeil?

File: cockatrice/src/userinfobox.cpp [hunk 2]
```diff
@@ -131,7 +130,7 @@ void UserInfoBox::updateInfo(const ServerInfo_User &user)
         if (user.accountage_secs() == 0)
             accountAgeString = tr("Unknown");
         else {
-            qint64 seconds = user.accountage_secs();
+            auto seconds = static_cast<qint64>(user.accountage_secs());
```
Comment (ebbit1q) [inline cockatrice/src/userinfobox.cpp:133]: is this necessary?

File: cockatrice/src/userinfobox.cpp [hunk 3]
```diff
@@ -305,5 +304,6 @@ void UserInfoBox::resizeEvent(QResizeEvent *event)
 {
     QPixmap resizedPixmap = avatarPixmap.scaled(avatarLabel.size(), Qt::KeepAspectRatio, Qt::SmoothTransformation);
     avatarLabel.setPixmap(resizedPixmap);
+    resize(sizeHint());
```
Comment (ebbit1q) [inline cockatrice/src/userinfobox.cpp:307]: resizing the window will result in an infinite recursion now....

File: cockatrice/src/userinfobox.cpp (hunk 1 above)
Comment (ebbit1q) [reply cockatrice/src/userinfobox.cpp:27]: actually, why not use setPointSizeF?

Comment (ZeldaZach) [reply cockatrice/src/userinfobox.cpp:27]: Didn't know about it -- changed

File: cockatrice/src/userinfobox.cpp (hunk 2 above)
Comment (ZeldaZach) [reply cockatrice/src/userinfobox.cpp:132]: It was a warning that was simple enough to handle as it's an unsigned to signed conversion

Comment (ebbit1q) [reply cockatrice/src/userinfobox.cpp:132]: why not use unsigned then

File: cockatrice/src/userinfobox.cpp [hunk 4]
```diff
@@ -24,7 +23,7 @@ UserInfoBox::UserInfoBox(AbstractClient *_client, bool _editable, QWidget *paren
 {
     QFont nameFont = nameLabel.font();
     nameFont.setBold(true);
-    nameFont.setPointSize(nameFont.pointSize() * 1.5);
+    nameFont.setPointSizeF(nameFont.pointSize() * 1.5);
```
Comment (ebbit1q) [inline cockatrice/src/userinfobox.cpp:26]:
```suggestion
    nameFont.setPointSizeF(nameFont.pointSizeF() * 1.5);
```

**Comment by ebbit1q** [comment]
wait, why do we do something like assigning SIXTY as a constant, like what

File: cockatrice/src/userinfobox.cpp [hunk 5]
```diff
@@ -11,20 +11,19 @@
...
-#include <QHBoxLayout>
 #include <QLabel>
 #include <QMessageBox>
 
-const qint64 SIXTY = 60;
-const qint64 HOURS_IN_A_DAY = 24;
-const qint64 DAYS_IN_A_YEAR = 365;
+const quint64 SIXTY = 60;
+const quint64 HOURS_IN_A_DAY = 24;
+const quint64 DAYS_IN_A_YEAR = 365;
```
Comment (tooomm) [inline cockatrice/src/userinfobox.cpp:19]: >wait, why do we do something like assigning SIXTY as a constant, like what

I assume the intention was to name it `MINUTES_IN_AN_HOUR` when looking at the other constant names.

**Comment by ebbit1q** [comment]
I made a pr that will address this #4526

**Pull request merged**